lazyflow/utility/reorderAxesDecorator.py

```diff
@@ -0,0 +1,447 @@
+            self.lazyOp2.in_b.connect(self.lazyOp1.out_b)
+
+        def check_lazyOp1(self):
+            # check that method types are treated correctly (actual tests in LazyOp)
+            self.lazyOp1.meth(*test_args)
+            self.lazyOp1.class_meth(*test_args)
+            self.lazyOp1.static_meth(*test_args)
+            self.lazyOp1.property_meth
+            self.lazyOp1.except_meth(*test_args)
+
+            # when accessing the child's operators they should conform with the outside order. Here, the shape of the
+            # resulting numpy.ndarray is tested to make sure we are not just checking meta data.
+            assert self.lazyOp1.out_a[:].wait().shape == outer_shape
+            assert self.lazyOp1.out_b[:].wait().shape == outer_shape
+
+            tags = vigra.defaultAxistags(outer_order)
+            expect = vigra.VigraArray(outer_shape, value=3, axistags=tags)
+
+            assert (expect == self.lazyOp1.out_b[:].wait()).all()
+            assert expect.axistags == self.lazyOp1.out_b.meta.axistags
+            assert expect.shape == self.lazyOp1.out_b.meta.shape
+
+        def check_lazyOp2(self):
+            assert self.lazyOp2.in_a[:].wait().shape == outer_shape
+            assert self.lazyOp2.in_b[:].wait().shape == outer_shape
+            assert self.lazyOp2.out_a[:].wait().shape == outer_shape
+            assert self.lazyOp2.out_b[:].wait().shape == outer_shape
+
+            tags = vigra.defaultAxistags(outer_order)
+            expect = vigra.VigraArray(outer_shape, value=4, axistags=tags)
+            assert (expect == self.lazyOp2.out_b[:].wait()).all()
+            assert expect.axistags == self.lazyOp2.out_b.meta.axistags
+            assert expect.shape == self.lazyOp2.out_b.meta.shape
+
+            # checking inner shapes by pretending to make inner calls
+            self.lazyOp2._inner_call = True
+            assert self.lazyOp2.in_a[:].wait().shape == inner_shape
+            assert self.lazyOp2.in_b[:].wait().shape == inner_shape
+            assert self.lazyOp2.out_a[:].wait().shape == inner_shape
+            assert self.lazyOp2.out_b[:].wait().shape == inner_shape
+
+    # ensemble of two example operators
+    opParent = OpParent(graph=Graph())
+    opParent.setupOutputs()
+    opParent.check_lazyOp1()
+    opParent.check_lazyOp2()
+    generateSvgFileForOperator(os.path.join(svg_dir, 'opParent.svg'), opParent, 5)
+
+    # wrapped test operator
+    lazyWrap = OperatorWrapper(LazyOp, 
+                               broadcastingSlotNames=['in_a', 'in_b'],
+                               operator_kwargs={'inner_order': inner_order}, 
+                               parent=opParent)
+    lazyWrap.out_a.resize(2)
+
+    tags = vigra.defaultAxistags(outer_order)
+    a = vigra.VigraArray(outer_shape, value=1, axistags=tags)
+    b = vigra.VigraArray(outer_shape, value=2, axistags=tags)
+    lazyWrap.in_a.setValue(a)
+    lazyWrap.in_b.setValue(b)    
```

Comment:
trailing spaces

Reply:
shame... btw, One 'trailing spaces' comment per reviewed file will do the trick in the future ;-)